File: bin/lib/prompts.js
'use strict';

const readline = require('node:readline');

function createRL() {
  return readline.createInterface({ input: process.stdin, output: process.stderr });
}

async function ask(question, defaultValue) {
  const rl = createRL();
  const hint = defaultValue !== undefined && defaultValue !== null && defaultValue !== ''
    ? ` [${defaultValue}]`
    : '';
  try {
    return await new Promise((resolve) => {
      rl.question(`? ${question}${hint}: `, (answer) => {
        const v = (answer || '').trim();
        resolve(v || (defaultValue ?? ''));
      });
    });
  } finally {
    rl.close();
  }
}

async function askYesNo(question, defaultYes) {
  const hint = defaultYes ? '[Y/n]' : '[y/N]';
  const rl = createRL();
  try {
    return await new Promise((resolve) => {
      rl.question(`? ${question} ${hint}: `, (answer) => {
        const a = (answer || '').trim().toLowerCase();
        if (a === '') return resolve(!!defaultYes);
        if (['y', 'yes', '1', 'true'].includes(a)) return resolve(true);
        if (['n', 'no', '0', 'false'].includes(a)) return resolve(false);
        // Re-prompt by recursing once
        resolve(askYesNo(question, defaultYes));
      });
    });
  } finally {
    rl.close();
  }
}

module.exports = { ask, askYesNo };

File: agentpress_cli/_prompts.py
"""Interactive prompts. Mirrors bin/lib/prompts.js."""
from __future__ import annotations

import sys


def ask(question: str, default: str | None = None) -> str:
    hint = f" [{default}]" if default else ""
    sys.stderr.write(f"? {question}{hint}: ")
    sys.stderr.flush()
    try:
        answer = input().strip()
    except EOFError:
        return default or ""
    return answer or (default or "")


def ask_yes_no(question: str, default_yes: bool = True) -> bool:
    hint = "[Y/n]" if default_yes else "[y/N]"
    while True:
        sys.stderr.write(f"? {question} {hint}: ")
        sys.stderr.flush()
        try:
            a = input().strip().lower()
        except EOFError:
            return bool(default_yes)
        if a == "":
            return bool(default_yes)
        if a in {"y", "yes", "1", "true"}:
            return True
        if a in {"n", "no", "0", "false"}:
            return False
